vortex-api/src/main/java/mil/army/usace/hec/vortex/temporal/TemporalDataCalculator.java

```diff
@@ -0,0 +1,243 @@
+        // Utility Class
+    }
+
+    public static float[] calculateWeightedAccumulation(List<VortexGrid> grids, ZonedDateTime startTime, ZonedDateTime endTime) {
+        // Accumulation Data = [Sum of grids' (data * overlap ratio)]
+        if (hasNoTargetData(grids, startTime, endTime)) return undefinedGridData();
+        double[] weights = calculateAccumulationWeights(grids, startTime, endTime);
+        return calculateWeightedData(grids, weights);
+    }
+
+    public static float[] calculateWeightedAverage(List<VortexGrid> grids, ZonedDateTime startTime, ZonedDateTime endTime) {
+        // Average Data = [Sum of grids' (data * overlap ratio)] / [total overlap ratio]
+        if (hasNoTargetData(grids, startTime, endTime)) return undefinedGridData();
+        double[] weights = calculateAverageWeights(grids, startTime, endTime);
+        return calculateWeightedData(grids, weights);
+    }
+
+    public static float[] calculatePointInstant(List<VortexGrid> vortexGrids, ZonedDateTime time) {
+        // Instant Data = floor grid + (rate of change between floor and ceiling) * (overlap time)
+        if (hasNoTargetData(vortexGrids, time, time)) return undefinedGridData();
+        double[] weights = calculatePointInstantWeights(vortexGrids, time);
+        return calculateWeightedData(vortexGrids, weights);
+    }
+    
+    public static float[] calculatePeriodInstant(List<VortexGrid> grids, ZonedDateTime startTime, ZonedDateTime endTime) {
+        if (hasNoTargetData(grids, startTime, endTime)) return undefinedGridData();
+        float[] result = new float[grids.get(0).data().length];
+
+        ZonedDateTime remainingStart = startTime;
+
+        for (int i = 0; i < grids.size() - 1; i++) {
+            VortexGrid grid1 = grids.get(i);
+            VortexGrid grid2 = grids.get(i + 1);
+
+            ZonedDateTime avgEnd = grid2.startTime();
+            double[] avgData = calculateAverageOfTwoGrids(grid1, grid2, remainingStart, endTime);
+
+            for (int j = 0; j < avgData.length; j++) {
+                result[j] = (float) (result[j] + avgData[j]);
+            }
+
+            remainingStart = avgEnd;
+        }
+
+        int totalInterval = (int) Duration.ofSeconds(endTime.toEpochSecond() - startTime.toEpochSecond()).toMinutes();
+        for (int i = 0; i < result.length; i++) {
+            result[i] /= totalInterval;
+        }
+
+        return result;
+    }
+
+    public static float[][] getMinMaxForGrids(List<VortexGrid> grids) {
+        int dataLength = grids.get(0).data().length;
+
+        float[][] minMaxData = new float[2][dataLength];
+        Arrays.fill(minMaxData[0], Float.MAX_VALUE);
+        Arrays.fill(minMaxData[1], -Float.MAX_VALUE);
+
+        for (VortexGrid grid : grids) {
+            float[] currentData = grid.data();
+            for (int i = 0; i < dataLength; i++) {
+                float datum = currentData[i];
+
+                if (grid.isNoDataValue(datum))
```

Comment:
Going through the method every time is more computationally expensive than a `==` check, right?

Reply:
I think the difference would be negligible, since `isNoDataValue` checks for `== Heclib.UNDEFINED_DOUBLE` and `Double.isNaN`, which checks for `== NaN`.